assets/js/i18n/locales.js

```diff
@@ -0,0 +1,45 @@
+import { t } from '@lingui/macro'
+// import { csrfToken } from '../utils'
+
+const supportedLocales = {
+  en: t`English`,
+  el: t`Greek`
+}
+
+const defaultLocale = 'en'
+
+const currentLocale = window.i18n && window.i18n.locale && isSupported(window.i18n.locale) ? window.i18n.locale : defaultLocale
+
+function isSupported (locale) {
+  return Object.prototype.hasOwnProperty.call(supportedLocales, locale)
+}
+
+/**
+ * This function tries to change locale by reloading the page and adding a locale query parameter
+ */
+function setLocaleWithQueryParameter (locale) {
+  if (isSupported(locale)) {
```

Comment:
If `isSupported()` returns false here, then nothing happens, which I think is 

- conceptually valid ("if you request to change the locale to something that we can't change to, we ignore you")
- better than reloading the page with some default locale

Reply:
Please phrase the user story. How does the user request to change the locale to something that doesn't exist?

Reply:
Indeed, no (benevolent) user will be able to try to change the locale to something that doesn't exist. However, the function `setLocale` by itself can be agnostic of this fact.

Reply:
Here are two conventions in the Workbench codebase:

1. Don't validate input arguments unless you're helping a human (developer or user) fix his or her own mistake.
2. If you're validating input arguments and you find an invalid argument, send that message back to the human. If the human is a developer, crash _hard_. (In JavaScript, that means `throw new Error(...)`. In Python, raise an Exception.) If the human is a user, help the user correct his or her mistake.

When you set `currentLocale` higher up in this file, you auto-select `en` instead of allowing the user to input something wrong (such as `xxxxx`). That's fine: clearly there is a human who can make a mistake; and while prompting the user might be nice, it takes _less code_ to select "en".

On this line, I don't see the story of a human making a mistake. And if there _were_ a story of a human developer calling this function incorrectly, then we'd want to help the human developer correct his or her mistake, rather than do nothing. Deleting this if-statement helps the human figure out what went wrong.